decoder/vaapidecoder_h264.cpp

```diff
@@ -1637,6 +1647,19 @@ YamiStatus VaapiDecoderH264::ensureContext(const SharedPtr<SPS>& sps)
             return status;
         // return YAMI_DECODE_FORMAT_CHANGE to info upper layer va context changed
         return YAMI_DECODE_FORMAT_CHANGE;
+    }else if(isResolutionChanged(sps)){
+        m_configBuffer.width = sps->frame_cropping_flag ? sps->m_cropRectWidth
```

Comment:
@wudping ,  maybe you can set them fristly: bool isResChanged = isResolutionChanged(sps); bool isContextChanged = (isDecodeContextChanged(sps); And don't forget to verify the attached clip test.zip can work now.